koku/api/provider/models.py

```diff
@@ -275,7 +264,7 @@ def _set_infrastructure_id_null(self):
         # can be shared among other providers, we must set the api_provider.infrastructure_id to null
         # for the infrastructure_id targeted for delete that is shared across other providers.
         # That is a confusing sentence to describe a circular relation between tables.
-        _sql = f"""
+        _sql = """
```

Comment:
It's always bugged me how Python makes it hard to store multi-line blocks of text at the same indentation level of the containing method or function. One approach I've used is using `textwrap.dedent()`, which was added in Python 3.3.

```suggestion
        _sql = """\
            update public.api_provider p
                set infrastructure_id = %s
               from public.api_providerinfrastructuremap m
             where m.infrastructure_provider_id = %s::uuid
               and p.infrastructure_id = m.id
            ;\
        """ 
        
        cur.execute(textwrap.dedent(_sql), params)
```

The `# nova: F541` on line 274 can be removed.

Reply:
I don't particularly like how this is formatted either, but this code is written so you can copy it into a sql file or into the terminal without dealing with formatting.

Reply:
Oh that's handy. I hadn't thought about that.